### PR TITLE
vim bump

### PR DESCRIPTION
**Pull request by ezzieyguywuf**
- app-editors/vim-core: bump to 9.0.0363
- app-editors/vim: bump to 9.0.0363
- app-editors/gvim: bump to 9.0.363

### CODE REVIEW

File: app-editors/vim-core/vim-core-9.0.0363.ebuild [hunk 1]
```diff
@@ -0,0 +1,230 @@
...
+inherit bash-completion-r1 desktop flag-o-matic prefix toolchain-funcs vim-doc xdg-utils
+
+if [[ ${PV} == 9999* ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/vim/vim.git"
+	EGIT_CHECKOUT_DIR=${WORKDIR}/vim-${PV}
+else
+	SRC_URI="https://github.com/vim/vim/archive/v${PV}.tar.gz -> ${P}.tar.gz
+		https://gitweb.gentoo.org/proj/vim-patches.git/snapshot/vim-patches-vim-9.0.0049-patches.tar.gz"
+	KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~loong ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~x64-cygwin ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
```
Comment (laumann) [inline app-editors/vim-core/vim-core-9.0.0363.ebuild:18]: I think you need to drop all keywords to ~arch

File: app-editors/gvim/gvim-9.0.0363.ebuild [hunk 2]
```diff
@@ -0,0 +1,361 @@
...
+inherit bash-completion-r1 flag-o-matic lua-single prefix python-single-r1 ruby-single toolchain-funcs vim-doc xdg-utils
+
+if [[ ${PV} == 9999* ]]; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/vim/vim.git"
+	EGIT_CHECKOUT_DIR=${WORKDIR}/vim-${PV}
+else
+	SRC_URI="https://github.com/vim/vim/archive/v${PV}.tar.gz -> vim-${PV}.tar.gz
+		https://gitweb.gentoo.org/proj/vim-patches.git/snapshot/vim-patches-vim-9.0.0049-patches.tar.gz"
+	KEYWORDS="~alpha amd64 arm arm64 ~hppa ~ia64 ppc ppc64 ~riscv ~s390 sparc x86 ~amd64-linux ~x86-linux ~ppc-macos ~x86-solaris"
```
Comment (laumann) [inline app-editors/gvim/gvim-9.0.0363.ebuild:23]: drop to ~arch

File: app-editors/vim/vim-9.0.0363.ebuild [hunk 3]
```diff
@@ -0,0 +1,357 @@
...
+
+inherit bash-completion-r1 desktop flag-o-matic lua-single python-single-r1 ruby-single toolchain-funcs vim-doc xdg-utils
+
+if [[ ${PV} == 9999* ]] ; then
+	inherit git-r3
+	EGIT_REPO_URI="https://github.com/vim/vim.git"
+else
+	SRC_URI="https://github.com/vim/vim/archive/v${PV}.tar.gz -> ${P}.tar.gz
+		https://gitweb.gentoo.org/proj/vim-patches.git/snapshot/vim-patches-vim-9.0.0049-patches.tar.gz"
+	KEYWORDS="~alpha amd64 arm arm64 hppa ~ia64 ~loong ~m68k ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~x64-cygwin ~amd64-linux ~x86-linux ~ppc-macos ~x64-macos ~sparc-solaris ~sparc64-solaris ~x64-solaris ~x86-solaris"
```
Comment (laumann) [inline app-editors/vim/vim-9.0.0363.ebuild:22]: drop to ~arch

**Comment by laumann** [comment]
Looks fine to me, from what I could tell:

 - PDEPEND on app-vim/gentoo-syntax is moved from vim-core to (g)vim
 - sorted the inherit lines (I considered if EXPORTED_FUNCTIONS might be problematic, but most of the phase functions are defined in all the ebuilds and most of the eclasses used are utility ones that don't define phase afaict)
 - nice catch with the implicit dependency on toolchain-funcs (tc-is-cross-compiler)

**Comment by ajakk** [comment]
If it's trivial to do so, can you bump up to 9.0.0389 and add a `Bug: https://bugs.gentoo.org/861092`?

**Comment by ezzieyguywuf** [comment]
@laumann thanks for taking a look. I can't really take credit for the `tc-is-cross-compiler` thing, `pkgcheck` told me about it :-P

@ajakk bumping to `9.0.0389` shouldn't really be an issue, but I'm curious: what's special about this version? Does it close the referenced bug?

**Comment by ajakk** [comment]
Yes, that's exactly what's special about it.

https://bugs.gentoo.org/861092#c12:

"CVE-2022-3134 (https://huntr.dev/bounties/6ec79e49-c7ab-4cd6-a517-e7934c2eb9dc):

Use After Free in GitHub repository vim/vim prior to 9.0.0388.

Patch is in 9.0.0389..."

**Comment by ezzieyguywuf** [comment]
> If it's trivial to do so, can you bump up to 9.0.0389 and add a `Bug: https://bugs.gentoo.org/861092`?

Done

**Pull request closed without merging**